### DISCUSSION TITLE
How to npm install forked repo that is updated?

### DISCUSSION THREAD

**Discussion by chan-dev** [stephanrauh/ngx-extended-pdf-viewer#910] [Q&A]
I'm pretty new to installing a forked repo.

So far i've tried
1. Adding the `ngx-extended-pdf-viewer: git:https://link-to-forked-repo.git`
2. Local file: `ngx-extended-pdf-viewer: file:../path/embedded-pdf/ngx-extended-pdf-viewer`
3. Use `prepare: npm run build`
4. Install it directly via: `npm install https://github-link/ngx-extended-pdf-viewer.git`
It just throws an error
![install-fork](https://user-images.githubusercontent.com/46891115/132287428-93df8d35-a1eb-4f6c-88ac-0f7e5729e5d8.png)

But none of these worked so far.

How could i make it work, by just installing `npm install`?

**Comment by stephanrauh** [comment]
I'm simply putting all three projects into a common project folder:
![image](https://user-images.githubusercontent.com/3045767/132384507-030f79bc-672c-4f77-9c66-8f0996d77f1c.png)

I suppose you're trying to create a link between the showcase and the library. I believe that's possible, and I also believe that makes everything faster and smoother, but until now, I didn't do that. It's just three independent projects, loosely coupled by the scripts `./updateMozillasPdfViewer.sh` and `npm run showcase`.

**Comment by chan-dev** [comment]
Actually, I have a separate project that needs to consume the updated pdf.js. So my process was
1. Fork `ngx-extended-pdf-viewer`
2. Follow the custom build of pdf.js with ngx-extended-pdf-viewer
3. Make changes to pdf.js
4. Run `./updateMozillasPdfViewer.sh`
5. Push changes to the fork containing updates to pdf.js
6. switch to another project 
7. `npm install` the fork

What i really is the `ngx-extended-pdf-viewer` and the updated pdf.js
Basically, i want it to be installable into another project.
How do i fit the process for my use case?

**Reply by stephanrauh** [reply]
I guess your approach works, too. Maybe I should adopt it, too. When I started the project, I spent most of the time with ngx-extended-pdf-viewer, but nowadays most new features need to be implemented in my fork of pdf.js.

What I'm doing differently is step 5 and 7. Instead of running `npm install`, I simply copy the `dist` folder of the library - or, in your case - the build folder of pdf.js directly into the `node_modules` folder of my test application. 

If you're modifying pdf.js exclusively, it suffices to copy the pdf*.js files to the `node_modules/ngx-extended-pdf-viewer/bleeding-edge` folder of your target application. If you've also modified ngx-extended-pdf-viewer, it's best to use a modified version of `npm run showcase` or `npm run issue`.

Caveat: the filename of the `pdf*.js` file contains the build number. That's increased each time you commit your code. That version number, in turn, is tracked in `pdf-default-options.ts`.

**Reply by chan-dev** [reply]
Thanks for the response, but what would the approach since `node_modules` are essentially ignored by `.gitignore`.

**Reply by stephanrauh** [reply]
It's not necessary to push your changes. It suffices to overwrite the files of the `node_modules/ngx-extended-pdf-viewer/` folder of your target project.

(I feel this is one of the times when it's easier to show things than to describe them in written language...)

**Reply by chan-dev** [reply]
Thanks I see. Got it. I was thinking of it more of being automated but i guess this also works in my case.
In terms of showing things, do you mean screen recording?

**Reply by chan-dev** [reply]
Copying the compiled files worked. If i were to use the fork approach, what steps should i know or do, so when i use `npm install`, it will work?

At the end of the day, what i want to achieve is based from our previous discussion.
Disabling `file` query parameter.
https://github.com/stephanrauh/ngx-extended-pdf-viewer/discussions/904#discussioncomment-1287317

**Reply by stephanrauh** [reply]
Awesome! Actually, I didn't think of a screen recording. I just thought we're talking at cross-purposes. At least I hope that's the correct English translation. :) I felt we'd be able to solve the problem in five minutes if we'd happen to sit in the same room, but sometimes it's difficult to explain simple ideas using written speech. Be that as it may, you've found a solution, and that's great!

**Reply by chan-dev** [reply]
Yeah, although it worked, i'm looking for a way to automate this and hopefully make it work using a `fork`. Let's say i'll just fork this repo and make changes and commit to it. How do i just install it? I believe some repo have some `prepare` npm scripts first or similar scripts, but it seems like the one under `projects/ngx-extended-pdf-viewer` is the one being used by the consuming app.
